cylc/flow/xtriggers/workflow_state.py

```diff
@@ -14,101 +14,93 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-import sqlite3
-from typing import Dict, Optional, Tuple
+from typing import Dict, Optional, Tuple, Any
+import asyncio
 
-from metomi.isodatetime.parsers import TimePointParser
-
-from cylc.flow.cycling.util import add_offset
-from cylc.flow.dbstatecheck import CylcWorkflowDBChecker
-from cylc.flow.pathutil import get_cylc_run_dir
-from cylc.flow.workflow_files import infer_latest_run_from_id
+from cylc.flow.scripts.workflow_state import WorkflowPoller
+from cylc.flow.id import tokenise
+from cylc.flow.exceptions import WorkflowConfigError
+from cylc.flow.task_state import TASK_STATUS_SUCCEEDED
 
 
 def workflow_state(
-    workflow: str,
-    task: str,
-    point: str,
+    workflow_task_id: str,
```

Comment:
This seems like a massive breaking change...

Could you not have something like

```python
def workflow_state(*args, **kwargs):
    if len(args) == 1 or 'workflow_task_id' in kwargs:
        _workflow_state()
    else:
        LOG.warn('using back compatibility version of xtrigger')
        _workflow_state_compat()
```

Reply:
I decided a breaking change was preferable because full backward compatibility is not possible. 

The main reason is the old command allows both `--output` and `--message` but, contrary to the help docs, they both query the task message. Whereas now, post optional outputs, "output" means the trigger name, not the task message.

However, I just checked again and the xtrigger function (unlike its CLI counterpart) doesn't have an alternative `output` arg, so maybe I can do it for that...

Reply:
This is no longer a breaking change - I worked around it.